Source/View Controllers/UploadViewController.swift

```diff
@@ -218,9 +239,29 @@ class UploadViewController: NSViewController {
             && (payloadName != nil)
             && !payloadName.isEmpty
             && (payloadIdentifier != nil)
-            && !payloadIdentifier.isEmpty) else { return }
+            && !payloadIdentifier.isEmpty
+            && isSiteReadyToUpload())
+            else { return }
+
         readyForUpload = !readyForUpload
     }
+
+    func isSiteReadyToUpload() -> Bool {
+        if site {
+            if let siteId = siteId, let siteName = siteName, siteId != "", siteName != "" {
+                return true
+            } else {
+                return false
+            }
+        } else {
+            return true
+        }
+    }
+
+    func updateSiteUI() {
```

Comment:
Since this project makes heavy use of bindings, if you wanted to you could remove this hide/show logic from the code and use bindings on the labels' Hidden property, and tie it to the Site checkbox state.